### PR TITLE
Refactor test modules

### PR DESCRIPTION
**Pull request by moarychan**
Fixes Azure/azure-sdk-for-java#23773

Changelog:
* Add [testcase document for Spring Cloud Azure](https://github.com/moarychan/azure-sdk-for-java/blob/moary/refactor-test-modules/sdk/spring/testcase-document.adoc#spring-cloud-azure-test-cases-scenarios) , will put this file in another place, delete first.
* Refactor testcases

  Refactor prograss
- [x] Actuator IT
- [x] Cosmos  IT
- [x] Keyvault IT
- [x] Keyvault Certificates IT
- [x] Service Bus JMS IT
- [x] Storage IT
- [x] Service Bus IT & EventHubs IT. Binder's UTs still need to be processed
- [ ] ~~Delete test modules~~, deleted `spring-cloud-azure-test-servicebus-binder`, other moduels will be removed in another PR.

### CODE REVIEW

File: sdk/spring/spring-cloud-azure-stream-binder-eventhubs/src/test/java/com/azure/spring/cloud/stream/binder/eventhubs/EventHubPartitionBinderTests.java [hunk 1]
```diff
@@ -1,81 +1,165 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-//package com.azure.spring.cloud.stream.binder.eventhubs;
```
Comment (stliu) [inline sdk/spring/spring-cloud-azure-stream-binder-eventhubs/src/test/java/com/azure/spring/cloud/stream/binder/eventhubs/EventHubPartitionBinderTests.java:4]: delete

File: sdk/spring/spring-cloud-azure-stream-binder-eventhubs/src/test/java/com/azure/spring/cloud/stream/binder/eventhubs/config/EventHubsBinderConfigurationTests.java [hunk 2]
```diff
@@ -1,32 +1,40 @@
...
+            .run(context -> {
+                assertThat(context).doesNotHaveBean(EventHubsBinderConfiguration.class);
+                assertThat(context).doesNotHaveBean(EventHubsMessageChannelBinder.class);
+            });
+    }
+
     @Test
     void shouldConfigureDefaultChannelProvisionerWhenNoResourceManagerProvided() {
         this.contextRunner
-            // TODO (xiada) this EventHubsProducerFactory should be deleted after health indicator refactoring
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-stream-binder-eventhubs/src/test/java/com/azure/spring/cloud/stream/binder/eventhubs/config/EventHubsBinderConfigurationTests.java:28]: cool

File: sdk/spring/spring-integration-azure-core/src/test/java/com/azure/spring/integration/handler/MessageHandlerTest.java [hunk 3]
```diff
@@ -1,130 +0,0 @@
-// Copyright (c) Microsoft Corporation. All rights reserved.
-// Licensed under the MIT License.
-
-package com.azure.spring.integration.handler;
```
Comment (saragluna) [inline sdk/spring/spring-integration-azure-core/src/test/java/com/azure/spring/integration/handler/MessageHandlerTest.java:4]: Why is this deleted instead of renaming?

File: sdk/spring/spring-integration-azure-servicebus/src/test/java/com/azure/spring/integration/servicebus/inbound/ServiceBusInboundChannelAdapterTests.java [hunk 4]
```diff
@@ -24,8 +24,8 @@
 
 import static org.mockito.Mockito.mock;
 
-public class ServiceBusInboundAdapterTest {
-    //TODO(yiliu6): extends from InboundChannelAdapterTest
+public class ServiceBusInboundChannelAdapterTests {
```
Comment (saragluna) [inline sdk/spring/spring-integration-azure-servicebus/src/test/java/com/azure/spring/integration/servicebus/inbound/ServiceBusInboundChannelAdapterTests.java:27]: Do we still need this TODO?

**Review by saragluna — CHANGES_REQUESTED** [review]

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/eventhubs/AzureEventHubsClientBuilderConfiguration.java [hunk 5]
```diff
@@ -24,7 +24,7 @@
 @Configuration(proxyBeanMethods = false)
 @ConditionalOnAnyProperty(prefix = "spring.cloud.azure.eventhubs", name = { "connection-string", "namespace" })
 @ConditionalOnProperty(prefix = "spring.cloud.azure.eventhubs", name = "event-hub-name")
-class AzureEventHubsClientBuilderConfiguration {
+public class AzureEventHubsClientBuilderConfiguration {
```
[the remark this answers is not in the document]
Comment (moarychan) [reply sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/eventhubs/AzureEventHubsClientBuilderConfiguration.java:27]: Test case uses it, https://github.com/Azure/azure-sdk-for-java/blob/8a384ae5e3edc397e5dac46369afebe301b8edf5/sdk/spring/spring-cloud-azure-actuator-autoconfigure/src/test/java/com/azure/spring/cloud/actuate/autoconfigure/eventhubs/EventHubsHealthConfigurationTests.java#L28
```text
NoUniqueBeanDefinitionException: No qualifying bean of type 'com.azure.spring.service.eventhubs.factory.EventHubClientBuilderFactory' available: expected single matching bean but found 3: eventHubClientBuilderFactory,EVENT_HUB_CONSUMER_CLIENT_BUILDER_FACTORY_BEAN_NAME,EVENT_HUB_PRODUCER_CLIENT_BUILDER_FACTORY_BEAN_NAME
```

File: sdk/spring/spring-integration-azure-core/src/test/java/com/azure/spring/integration/handler/MessageHandlerTest.java (hunk 3 above)
Comment (moarychan) [reply sdk/spring/spring-integration-azure-core/src/test/java/com/azure/spring/integration/handler/MessageHandlerTest.java:4]: I did refactoring, and sometimes it just happened.

File: sdk/spring/spring-integration-azure-servicebus/src/test/java/com/azure/spring/integration/servicebus/inbound/ServiceBusInboundChannelAdapterTests.java (hunk 4 above)
Comment (saragluna) [reply sdk/spring/spring-integration-azure-servicebus/src/test/java/com/azure/spring/integration/servicebus/inbound/ServiceBusInboundChannelAdapterTests.java:27]: Maybe we can remove these two classes AbstractInboundChannelAdapter, InboundChannelAdapterTest?

File: sdk/spring/spring-integration-azure-core/src/test/java/com/azure/spring/integration/handler/MessageHandlerTest.java (hunk 3 above)
Comment (saragluna) [reply sdk/spring/spring-integration-azure-core/src/test/java/com/azure/spring/integration/handler/MessageHandlerTest.java:4]: Do you think it's possible for you to bring the git history back?

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/eventhubs/AzureEventHubsClientBuilderConfiguration.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (saragluna) [reply sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/eventhubs/AzureEventHubsClientBuilderConfiguration.java:27]: Maybe in the `EventHubsHealthConfigurationTests.java` we can provide a bean instead of calling the AzureEventHubsClientBuilderConfiguration class?

[the remark this answers is not in the document]
Comment (moarychan) [reply sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/eventhubs/AzureEventHubsClientBuilderConfiguration.java:27]: Good idea!

File: sdk/spring/spring-integration-azure-servicebus/src/test/java/com/azure/spring/integration/servicebus/inbound/ServiceBusInboundChannelAdapterTests.java (hunk 4 above)
Comment (moarychan) [reply sdk/spring/spring-integration-azure-servicebus/src/test/java/com/azure/spring/integration/servicebus/inbound/ServiceBusInboundChannelAdapterTests.java:27]: OK
